### PR TITLE
Some changes suggested by lummox

### PR DESCRIPTION
**Pull request by optimumtact**
We need to remove the client immediately otherwise it can still spam keysends until the timer fires.

Also that other code just has poor performance for no reason

### CODE REVIEW

File: code/modules/keybindings/bindings_client.dm [hunk 1]
```diff
@@ -20,22 +20,22 @@
...
 	if(cache >= MAX_KEYPRESS_AUTOKICK)
 		if(!keysend_tripped)
-			keysend_tripped = TRUE		
+			keysend_tripped = TRUE
 			next_keysend_trip_reset = world.time + (2 SECONDS)
 		else
 			log_admin("Client [ckey] was just autokicked for flooding keysends; likely abuse but potentially lagspike.")
 			message_admins("Client [ckey] was just autokicked for flooding keysends; likely abuse but potentially lagspike.")
-			QDEL_IN(src, 1)
+			qdel(src)
```
Comment (optimumtact) [inline code/modules/keybindings/bindings_client.dm]: in case you're wondering client does return harddel now

**Comment by optimumtact** [comment]
@tgstation/commit-access

**Review by SpaceManiac — APPROVED** [review]
ergh, I dislike the RDSCREEN_NOBREAK design and it should have been changed long ago

changes look good to me

File: code/modules/keybindings/bindings_client.dm (hunk 1 above)
Comment (actioninja) [reply code/modules/keybindings/bindings_client.dm]: if it's harddelling couldn't that potentially be a vector as well? Feels exploitable.

Comment (optimumtact) [reply code/modules/keybindings/bindings_client.dm]: assuredly it would be hard to join fast enough to do that.

Comment (actioninja) [reply code/modules/keybindings/bindings_client.dm]: With a single account yeah, but we're not dealing with standard unmodified clients with normal limitations now.

**Pull request merged**